fuse_graphs/src/hash_graph.cpp

```diff
@@ -345,16 +353,32 @@ void HashGraph::getCovariance(
     throw std::runtime_error("Could not compute requested covariance blocks.");
   }
   // Populate the computed covariance blocks into the output variable.
-  // We use the temporary structure to avoid repeated map lookups.
-  for (size_t i = 0; i < covariance_requests.size(); ++i)
+  if (use_tangent_space)
+  {
+    for (size_t i = 0; i < covariance_requests.size(); ++i)
+    {
+      if (!covariance.GetCovarianceBlockInTangentSpace(all_covariance_blocks.at(i).first,
```

Comment:
Nit: maybe make const references to these to avoid the repeated indexing? Same for the other variables below.